BDArmory/Ammo/PooledBullet.cs

```diff
@@ -1707,30 +1705,34 @@ public static void FireBullet(BulletInfo bulletType, int projectileCount,
                 pBullet.isSubProjectile = isSubP;
                 pBullet.isAPSprojectile = isAPSP;
                 pBullet.gameObject.SetActive(true);
-                
-                if (isSubP)
-                    pBullet.SetTracerPosition();
 
                 if (registerShot)
                     BDACompetitionMode.Instance.Scores.RegisterShot(sourceInfo.vessel.GetName());
 
-                if (!pBullet.CheckBulletCollisions(timestep)) // Check that the bullet won't immediately hit anything and die.
+                if (!addSourcePartVel)
+                {
+                    pBullet.SetTracerPosition();
+                    if (!pBullet.CheckBulletCollisions(timestep)) continue; // Bullet immediately hit something and died.
+                    if (!pBullet.hasRicocheted) pBullet.MoveBullet(timestep); // Move the bullet the remaining part of the frame.
+                    pBullet.currentPosition += (TimeWarp.fixedDeltaTime - timestep) * BDKrakensbane.FrameVelocityV3f; // Re-adjust for Krakensbane.
+                    pBullet.timeAlive = timestep;
+                }
+                else if (!pBullet.CheckBulletCollisions(timestep)) // Check that the bullet won't immediately hit anything and die.
                 {
                     // The following gets bullet tracers to line up properly when at orbital velocities.
                     // It should be consistent with how it's done in Aim().
                     // Technically, there could be a small gap between the collision check and the start position, but this should be insignificant.
                     if (!pBullet.hasRicocheted) // Movement is handled internally for ricochets.
                     {
-                        var gravity = (Vector3)FlightGlobals.getGeeForceAtPosition(pBullet.currentPosition);
+                        Vector3 gravity = drop ? (Vector3)FlightGlobals.getGeeForceAtPosition(pBullet.currentPosition) : Vector3.zero;
                         pBullet.currentPosition = AIUtils.PredictPosition(pBullet.currentPosition, firedVelocity[i], gravity, timestep);
                         pBullet.currentVelocity += timestep * gravity; // Adjusting the velocity here mostly eliminates bullet deviation due to iTime.
                         pBullet.DistanceTraveled += timestep * pBullet.currentVelocity.magnitude; // Adjust the distance traveled to account for iTime.
                     }
-                    if (!BDKrakensbane.IsActive) pBullet.currentPosition += timestep * sourceInfo.weapon.rb.velocity; // If Krakensbane isn't active, bullets get an additional shift by this amount.
+                    if (addSourcePartVel && !BDKrakensbane.IsActive) pBullet.currentPosition += TimeWarp.fixedDeltaTime * sourceInfo.weapon.rb.velocity; // If Krakensbane isn't active, bullets get an additional shift by this amount.
```

Comment:
With using `if (!addSourcePartVel)` instead of `if (isSubP)` as the condition for the top branch, these extra `addSourcePartVel &&` conditions here and below aren't needed.